src/modules/formBuilder/components/itemEditor/RequiredOptionalRadio.tsx

```diff
@@ -0,0 +1,113 @@
+import { useCallback, useMemo } from 'react';
+import { useController, useWatch } from 'react-hook-form';
+import { FormItemControl } from './types';
+import RadioGroupInput from '@/components/elements/input/RadioGroupInput';
+import { PickListOption } from '@/types/gqlTypes';
+
+const options = [
+  {
+    code: 'required',
+    label: 'Required',
+  },
+  {
+    code: 'warnIfEmpty',
+    label: 'Warn If Empty',
+  },
+  {
+    code: 'optional',
+    label: 'Optional',
+  },
+  {
+    code: 'readOnly',
+    label: 'Read-Only',
+  },
+];
+
+interface Props {
+  control: FormItemControl;
+}
+
+const RequiredOptionalRadio: React.FC<Props> = ({ control }) => {
+  const { field: requiredField } = useController({
+    name: 'required',
+    control,
+    defaultValue: false,
+  });
+  const { field: warnIfEmptyField } = useController({
+    name: 'warnIfEmpty',
+    control,
+    defaultValue: false,
+  });
+  const { field: readOnlyField } = useController({
+    name: 'readOnly',
+    control,
+    defaultValue: false,
+  });
+
+  const [required, warnIfEmpty, readOnly] = useWatch({
+    control,
+    name: ['required', 'warnIfEmpty', 'readOnly'],
+  });
+
+  const radioValue = useMemo(() => {
+    if (required) {
+      return 'required';
+    } else if (warnIfEmpty) {
+      return 'warnIfEmpty';
+    } else if (readOnly) {
+      return 'readOnly';
+    } else {
+      return 'optional';
+    }
+  }, [readOnly, required, warnIfEmpty]);
+
+  const handleChange = useCallback(
+    (option?: PickListOption | null) => {
+      if (!option) return;
+
+      if (option.code === 'required') {
+        requiredField.onChange(true);
+        warnIfEmptyField.onChange(false);
+        readOnlyField.onChange(false);
+      } else if (option.code === 'warnIfEmpty') {
+        requiredField.onChange(false);
+        warnIfEmptyField.onChange(true);
+        readOnlyField.onChange(false);
+      } else if (option.code === 'readOnly') {
+        requiredField.onChange(false);
+        warnIfEmptyField.onChange(false);
+        readOnlyField.onChange(true);
+      } else if (option.code === 'optional') {
+        requiredField.onChange(false);
+        warnIfEmptyField.onChange(false);
+        readOnlyField.onChange(false);
+      }
+    },
+    [readOnlyField, requiredField, warnIfEmptyField]
+  );
+
+  return (
+    <RadioGroupInput
+      options={options}
+      label='Response Validation'
+      value={options.find((o) => o.code === radioValue)}
+      onChange={handleChange}
+      labelSx={{ typography: 'body2', color: 'text.secondary', pt: 1 }}
```

Comment:
👍 also I wonder if it would be nicer to display the helper text for each option below that option, rather than one big block below the whole item?

Reply:
Maybe, I pulled this approach from the designs. I think it would be possible to have extended labels on each radio element as another alternative. (But each long text would become the label for each element, not helper text).